numerics/polynomial_evaluators_body.hpp

```diff
@@ -118,126 +121,183 @@ struct InternalEstrinEvaluator<Value, Argument, degree, low, 0> {
-  return InternalEstrinEvaluator<Value, Argument, degree,
-                                 low, m - 1>::
-             EvaluateDerivative(coefficients, argument, argument_squares) +
-         std::get<n>(argument_squares) *
-             InternalEstrinEvaluator<Value, Argument, degree,
-                                     low + m, subdegree - m>::
-                 EvaluateDerivative(coefficients, argument, argument_squares);
+  auto const& xᵐ = std::get<n>(argument_squares);
+  auto const a =
+      InternalEstrinEvaluator<Value, Argument,
+                              degree, fma, low + m, subdegree - m>::
+          EvaluateDerivative(coefficients, argument, argument_squares);
+  auto const b =
+      InternalEstrinEvaluator<Value, Argument, degree, fma, low, m - 1>::
+          EvaluateDerivative(coefficients, argument, argument_squares);
+  if constexpr (fma) {
+    return FusedMultiplyAdd(a, xᵐ, b);
+  } else {
+    return a * xᵐ + b;
+  }
 }
 
-template<typename Value, typename Argument, int degree, int low>
-Derivative<Value, Argument, low>
-InternalEstrinEvaluator<Value, Argument, degree, low, 1>::Evaluate(
+template<typename Value, typename Argument, int degree, bool fma, int low>
+FORCE_INLINE(inline) Derivative<Value, Argument, low>
+InternalEstrinEvaluator<Value, Argument, degree, fma, low, 1>::Evaluate(
     Coefficients const& coefficients,
     Argument const& argument,
     ArgumentSquares const& argument_squares) {
-  return std::get<low>(coefficients) +
-         argument * std::get<low + 1>(coefficients);
+  auto const& x = argument;
+  auto const& a = std::get<low + 1>(coefficients);
+  auto const& b = std::get<low>(coefficients);
+  if constexpr (fma) {
+    return FusedMultiplyAdd(a, x, b);
+  } else {
+    return a * x + b;
+  }
 }
 
-template<typename Value, typename Argument, int degree, int low>
-Derivative<Value, Argument, low>
-InternalEstrinEvaluator<Value, Argument, degree, low, 0>::Evaluate(
+template<typename Value, typename Argument, int degree, bool fma, int low>
+FORCE_INLINE(inline) Derivative<Value, Argument, low>
+InternalEstrinEvaluator<Value, Argument, degree, fma, low, 0>::Evaluate(
     Coefficients const& coefficients,
     Argument const& argument,
     ArgumentSquares const& argument_squares) {
   return std::get<low>(coefficients);
 }
 
-template<typename Value, typename Argument, int degree, int low>
-Derivative<Value, Argument, low>
-InternalEstrinEvaluator<Value, Argument, degree, low, 1>::EvaluateDerivative(
-    Coefficients const& coefficients,
-    Argument const& argument,
-    ArgumentSquares const& argument_squares) {
-  return low * std::get<low>(coefficients) +
-         argument * (low + 1) * std::get<low + 1>(coefficients);
+template<typename Value, typename Argument, int degree, bool fma, int low>
+FORCE_INLINE(inline) Derivative<Value, Argument, low>
+InternalEstrinEvaluator<Value, Argument, degree, fma, low, 1>::
+    EvaluateDerivative(Coefficients const& coefficients,
```

Comment:
`EvaluateDerivative(` might fit on the previous line.

Reply:
Nope.